### DISCUSSION TITLE
what's the point of CRUD::setOperationSetting('withTrashFilter', false); if editing trashed users gives error 404?

### DISCUSSION THREAD

**Discussion by prescriptionlifeline** [Laravel-Backpack/community-forum#1255] [Q&A (Help)] [answered]
Quoting https://backpackforlaravel.com/docs/6.x/crud-operation-trash#how-to-configure-1 :

> You can easily disable the default trash filter:
> 
> ```php
> public function setupTrashOperation()
> {
>     CRUD::setOperationSetting('withTrashFilter', false);
> 
>     // one of the things the filter does is preventing the update/access to trashed items. 
>     // if you disable it and use Update/Show operations, you should manually disable the 
>     // access to those operations or they will throw 404:
>     CRUD::setAccessCondition(['update', 'show'], function($entry) {
>         return ! $entry->trashed();
>     });
> }
> ```
> Disabling the filter also will make the trashed items show in your List view.

What's the point of having trashed items in your List view if half of the actions that the List view provides (eg. Edit) yield an error 404 when you try to perform them?

Like even making the anonymous function in `CRUD::setAccessCondition` return `true` unconditionally does nothing. Even adding `edit` to the `['update', 'show']` bit does nothing.

So, again, I ask...  what's the point?

**Comment by prescriptionlifeline** [comment]
I looked into some more hackish approaches as well.

In vendor/backpack/crud/src/app/Library/CrudPanel/Traits/Read.php there's this:

```php
    public function getEntry($id)
    {
        if (! $this->entry) {
            if ($this->getOperationSetting('eagerLoadRelationships')) {
                $this->eagerLoadRelationshipFields();
            }
            $this->entry = $this->getModelWithCrudPanelQuery()->findOrFail($id);
            $this->entry = $this->entry->withFakes();
        }

        return $this->entry;
    }
```
Replacing `$this->getModelWithCrudPanelQuery()->findOrFail($id);` with `$this->entry = $this->getModelWithCrudPanelQuery()->withTrashed()->findOrFail($id);` makes it so that trashed users can be edited but modifying a vendor file has significant drawbacks. Since it's in a trait I thought maybe I could copy that method into my CRUD and modify it there as I see fit but that didn't seem to work.

In `setupShowOperation()` one can do `CRUD::set('show.softDeletes', true)` to _preview_ trashed entries. In vendor/backpack/crud/src/app/Library/CrudPanel/Traits/ShowOperation.php, in the `show()` method, there's this:

```php
        // get the info for that entry (include softDeleted items if the trait is used)
        if ($this->crud->get('show.softDeletes') && in_array('Illuminate\Database\Eloquent\SoftDeletes', class_uses($this->crud->model))) {
            $this->data['entry'] = $this->crud->getModel()->withTrashed()->findOrFail($id);
        } else {
            $this->data['entry'] = $this->crud->getEntryWithLocale($id);
        }
```
But I'm not seeing anything analogous to that in vendor/backpack/crud/src/app/Library/CrudPanel/Traits/UpdateOperation.php. I mean, in the `edit()` method there's `$this->data['entry'] = $this->crud->getEntryWithLocale($id);` so superficially one might think the `edit()` method could be copied to my CRUD but, oh wait, when I put in `exit('THIS FAR');` at the top of that method I can see that for a trashed user that method isn't even called so it wouldn't help, anyway.

So, again, my question still stands. What's the point of having trashed items in your List view if half of the actions that the List view provides (eg. Edit) yield an error 404 when you try to perform them?

**Comment by jcastroa87** [comment, marked as answer]
Hello @prescriptionlifeline 

The point of this functionality is to give developers more control over how trashed items are handled. Here are a few reasons why you might want to disable the default trash filter:

- Custom filtering: You might want to implement your own custom filtering logic for trashed items.
- Visibility of trashed items: In some cases, you might want to show trashed items alongside active items for comparison or reference.
- Custom access control: You can implement your own access control logic for trashed items.

Let me know if you have any other question.

Cheers.

**Reply by prescriptionlifeline** [reply]
> The point of this functionality is to give developers more control over how trashed items are handled. 

Well then maybe you could give developers _even more_ control over how trashed items are handled and let them make it so trashed items can be edited without requiring modifications to the files in the `vendor/` directory?
